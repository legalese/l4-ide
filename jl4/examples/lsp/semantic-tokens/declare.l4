ยง `Algebraic data types`

DECLARE List OF a
  IS ONE OF
    Nil
    Cons HAS
      head IS AN a
      tail IS A List OF a

DECLARE Tree a
  IS ONE OF
    Leaf HAS value IS AN a
    Node HAS left IS A Tree a
             right IS A Tree a

DECLARE Choice OF a, b
  IS ONE OF
    Left  HAS payload IS AN a
    Right HAS payload IS A  b

DECLARE PAIR OF a, b
  HAS fst IS AN a
      snd IS A  b

GIVEN a IS A TYPE
      b IS A TYPE
DECLARE PAIR2
  HAS fst IS AN a
      snd IS A  b

GIVEN a IS A TYPE
      b IS A TYPE
DECLARE PAIR3 a b
  HAS fst IS AN a
      snd IS A  b

GIVEN a IS A TYPE, b IS A TYPE
DECLARE PAIR4 a b HAS fst IS AN a, snd IS A  b

DECLARE Bool IS BOOLEAN

DECLARE MyList1 a IS LIST OF a

GIVEN a IS A TYPE
DECLARE MyList2 a IS LIST OF a
