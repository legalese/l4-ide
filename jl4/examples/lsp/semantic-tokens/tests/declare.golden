Phase: Lexer
1:1-2 SemanticTokenTypes_Operator
1:3-25 SemanticTokenTypes_Variable
3:1-8 SemanticTokenTypes_Keyword
3:9-13 SemanticTokenTypes_Variable
3:14-16 SemanticTokenTypes_Keyword
3:17-18 SemanticTokenTypes_Variable
4:3-5 SemanticTokenTypes_Keyword
4:6-9 SemanticTokenTypes_Keyword
4:10-12 SemanticTokenTypes_Keyword
5:5-8 SemanticTokenTypes_Variable
6:5-9 SemanticTokenTypes_Variable
6:10-13 SemanticTokenTypes_Keyword
7:7-11 SemanticTokenTypes_Variable
7:12-14 SemanticTokenTypes_Keyword
7:15-17 SemanticTokenTypes_Keyword
7:18-19 SemanticTokenTypes_Variable
8:7-11 SemanticTokenTypes_Variable
8:12-14 SemanticTokenTypes_Keyword
8:15-16 SemanticTokenTypes_Keyword
8:17-21 SemanticTokenTypes_Variable
8:22-24 SemanticTokenTypes_Keyword
8:25-26 SemanticTokenTypes_Variable
10:1-8 SemanticTokenTypes_Keyword
10:9-13 SemanticTokenTypes_Variable
10:14-15 SemanticTokenTypes_Variable
11:3-5 SemanticTokenTypes_Keyword
11:6-9 SemanticTokenTypes_Keyword
11:10-12 SemanticTokenTypes_Keyword
12:5-9 SemanticTokenTypes_Variable
12:10-13 SemanticTokenTypes_Keyword
12:14-19 SemanticTokenTypes_Variable
12:20-22 SemanticTokenTypes_Keyword
12:23-25 SemanticTokenTypes_Keyword
12:26-27 SemanticTokenTypes_Variable
13:5-9 SemanticTokenTypes_Variable
13:10-13 SemanticTokenTypes_Keyword
13:14-18 SemanticTokenTypes_Variable
13:19-21 SemanticTokenTypes_Keyword
13:22-23 SemanticTokenTypes_Keyword
13:24-28 SemanticTokenTypes_Variable
13:29-30 SemanticTokenTypes_Variable
14:14-19 SemanticTokenTypes_Variable
14:20-22 SemanticTokenTypes_Keyword
14:23-24 SemanticTokenTypes_Keyword
14:25-29 SemanticTokenTypes_Variable
14:30-31 SemanticTokenTypes_Variable
16:1-8 SemanticTokenTypes_Keyword
16:9-15 SemanticTokenTypes_Variable
16:16-18 SemanticTokenTypes_Keyword
16:19-20 SemanticTokenTypes_Variable
16:20-21 SemanticTokenTypes_Operator
16:22-23 SemanticTokenTypes_Variable
17:3-5 SemanticTokenTypes_Keyword
17:6-9 SemanticTokenTypes_Keyword
17:10-12 SemanticTokenTypes_Keyword
18:5-9 SemanticTokenTypes_Variable
18:11-14 SemanticTokenTypes_Keyword
18:15-22 SemanticTokenTypes_Variable
18:23-25 SemanticTokenTypes_Keyword
18:26-28 SemanticTokenTypes_Keyword
18:29-30 SemanticTokenTypes_Variable
19:5-10 SemanticTokenTypes_Variable
19:11-14 SemanticTokenTypes_Keyword
19:15-22 SemanticTokenTypes_Variable
19:23-25 SemanticTokenTypes_Keyword
19:26-27 SemanticTokenTypes_Keyword
19:29-30 SemanticTokenTypes_Variable
21:1-8 SemanticTokenTypes_Keyword
21:9-13 SemanticTokenTypes_Variable
21:14-16 SemanticTokenTypes_Keyword
21:17-18 SemanticTokenTypes_Variable
21:18-19 SemanticTokenTypes_Operator
21:20-21 SemanticTokenTypes_Variable
22:3-6 SemanticTokenTypes_Keyword
22:7-10 SemanticTokenTypes_Variable
22:11-13 SemanticTokenTypes_Keyword
22:14-16 SemanticTokenTypes_Keyword
22:17-18 SemanticTokenTypes_Variable
23:7-10 SemanticTokenTypes_Variable
23:11-13 SemanticTokenTypes_Keyword
23:14-15 SemanticTokenTypes_Keyword
23:17-18 SemanticTokenTypes_Variable
25:1-6 SemanticTokenTypes_Keyword
25:7-8 SemanticTokenTypes_Variable
25:9-11 SemanticTokenTypes_Keyword
25:12-13 SemanticTokenTypes_Keyword
25:14-18 SemanticTokenTypes_Keyword
26:7-8 SemanticTokenTypes_Variable
26:9-11 SemanticTokenTypes_Keyword
26:12-13 SemanticTokenTypes_Keyword
26:14-18 SemanticTokenTypes_Keyword
27:1-8 SemanticTokenTypes_Keyword
27:9-14 SemanticTokenTypes_Variable
28:3-6 SemanticTokenTypes_Keyword
28:7-10 SemanticTokenTypes_Variable
28:11-13 SemanticTokenTypes_Keyword
28:14-16 SemanticTokenTypes_Keyword
28:17-18 SemanticTokenTypes_Variable
29:7-10 SemanticTokenTypes_Variable
29:11-13 SemanticTokenTypes_Keyword
29:14-15 SemanticTokenTypes_Keyword
29:17-18 SemanticTokenTypes_Variable
31:1-6 SemanticTokenTypes_Keyword
31:7-8 SemanticTokenTypes_Variable
31:9-11 SemanticTokenTypes_Keyword
31:12-13 SemanticTokenTypes_Keyword
31:14-18 SemanticTokenTypes_Keyword
32:7-8 SemanticTokenTypes_Variable
32:9-11 SemanticTokenTypes_Keyword
32:12-13 SemanticTokenTypes_Keyword
32:14-18 SemanticTokenTypes_Keyword
33:1-8 SemanticTokenTypes_Keyword
33:9-14 SemanticTokenTypes_Variable
33:15-16 SemanticTokenTypes_Variable
33:17-18 SemanticTokenTypes_Variable
34:3-6 SemanticTokenTypes_Keyword
34:7-10 SemanticTokenTypes_Variable
34:11-13 SemanticTokenTypes_Keyword
34:14-16 SemanticTokenTypes_Keyword
34:17-18 SemanticTokenTypes_Variable
35:7-10 SemanticTokenTypes_Variable
35:11-13 SemanticTokenTypes_Keyword
35:14-15 SemanticTokenTypes_Keyword
35:17-18 SemanticTokenTypes_Variable
37:1-6 SemanticTokenTypes_Keyword
37:7-8 SemanticTokenTypes_Variable
37:9-11 SemanticTokenTypes_Keyword
37:12-13 SemanticTokenTypes_Keyword
37:14-18 SemanticTokenTypes_Keyword
37:18-19 SemanticTokenTypes_Operator
37:20-21 SemanticTokenTypes_Variable
37:22-24 SemanticTokenTypes_Keyword
37:25-26 SemanticTokenTypes_Keyword
37:27-31 SemanticTokenTypes_Keyword
38:1-8 SemanticTokenTypes_Keyword
38:9-14 SemanticTokenTypes_Variable
38:15-16 SemanticTokenTypes_Variable
38:17-18 SemanticTokenTypes_Variable
38:19-22 SemanticTokenTypes_Keyword
38:23-26 SemanticTokenTypes_Variable
38:27-29 SemanticTokenTypes_Keyword
38:30-32 SemanticTokenTypes_Keyword
38:33-34 SemanticTokenTypes_Variable
38:34-35 SemanticTokenTypes_Operator
38:36-39 SemanticTokenTypes_Variable
38:40-42 SemanticTokenTypes_Keyword
38:43-44 SemanticTokenTypes_Keyword
38:46-47 SemanticTokenTypes_Variable
40:1-8 SemanticTokenTypes_Keyword
40:9-13 SemanticTokenTypes_Variable
40:14-16 SemanticTokenTypes_Keyword
40:17-24 SemanticTokenTypes_Variable
42:1-8 SemanticTokenTypes_Keyword
42:9-16 SemanticTokenTypes_Variable
42:17-18 SemanticTokenTypes_Variable
42:19-21 SemanticTokenTypes_Keyword
42:22-26 SemanticTokenTypes_Keyword
42:27-29 SemanticTokenTypes_Keyword
42:30-31 SemanticTokenTypes_Variable
44:1-6 SemanticTokenTypes_Keyword
44:7-8 SemanticTokenTypes_Variable
44:9-11 SemanticTokenTypes_Keyword
44:12-13 SemanticTokenTypes_Keyword
44:14-18 SemanticTokenTypes_Keyword
45:1-8 SemanticTokenTypes_Keyword
45:9-16 SemanticTokenTypes_Variable
45:17-18 SemanticTokenTypes_Variable
45:19-21 SemanticTokenTypes_Keyword
45:22-26 SemanticTokenTypes_Keyword
45:27-29 SemanticTokenTypes_Keyword
45:30-31 SemanticTokenTypes_Variable

Phase: Parse
1:1-2 SemanticTokenTypes_Operator
1:3-25 SemanticTokenTypes_Interface
3:1-8 SemanticTokenTypes_Keyword
3:9-13 SemanticTokenTypes_Type
3:14-16 SemanticTokenTypes_Keyword
3:17-18 SemanticTokenTypes_TypeParameter
4:3-5 SemanticTokenTypes_Keyword
4:6-9 SemanticTokenTypes_Keyword
4:10-12 SemanticTokenTypes_Keyword
5:5-8 SemanticTokenTypes_Class
6:5-9 SemanticTokenTypes_Class
6:10-13 SemanticTokenTypes_Keyword
7:7-11 SemanticTokenTypes_Variable
7:12-14 SemanticTokenTypes_Keyword
7:15-17 SemanticTokenTypes_Keyword
7:18-19 SemanticTokenTypes_Type
8:7-11 SemanticTokenTypes_Variable
8:12-14 SemanticTokenTypes_Keyword
8:15-16 SemanticTokenTypes_Keyword
8:17-21 SemanticTokenTypes_Type
8:22-24 SemanticTokenTypes_Keyword
8:25-26 SemanticTokenTypes_Type
10:1-8 SemanticTokenTypes_Keyword
10:9-13 SemanticTokenTypes_Type
10:14-15 SemanticTokenTypes_TypeParameter
11:3-5 SemanticTokenTypes_Keyword
11:6-9 SemanticTokenTypes_Keyword
11:10-12 SemanticTokenTypes_Keyword
12:5-9 SemanticTokenTypes_Class
12:10-13 SemanticTokenTypes_Keyword
12:14-19 SemanticTokenTypes_Variable
12:20-22 SemanticTokenTypes_Keyword
12:23-25 SemanticTokenTypes_Keyword
12:26-27 SemanticTokenTypes_Type
13:5-9 SemanticTokenTypes_Class
13:10-13 SemanticTokenTypes_Keyword
13:14-18 SemanticTokenTypes_Variable
13:19-21 SemanticTokenTypes_Keyword
13:22-23 SemanticTokenTypes_Keyword
13:24-28 SemanticTokenTypes_Type
13:29-30 SemanticTokenTypes_Type
14:14-19 SemanticTokenTypes_Variable
14:20-22 SemanticTokenTypes_Keyword
14:23-24 SemanticTokenTypes_Keyword
14:25-29 SemanticTokenTypes_Type
14:30-31 SemanticTokenTypes_Type
16:1-8 SemanticTokenTypes_Keyword
16:9-15 SemanticTokenTypes_Type
16:16-18 SemanticTokenTypes_Keyword
16:19-20 SemanticTokenTypes_TypeParameter
16:20-21 SemanticTokenTypes_Operator
16:22-23 SemanticTokenTypes_TypeParameter
17:3-5 SemanticTokenTypes_Keyword
17:6-9 SemanticTokenTypes_Keyword
17:10-12 SemanticTokenTypes_Keyword
18:5-9 SemanticTokenTypes_Class
18:11-14 SemanticTokenTypes_Keyword
18:15-22 SemanticTokenTypes_Variable
18:23-25 SemanticTokenTypes_Keyword
18:26-28 SemanticTokenTypes_Keyword
18:29-30 SemanticTokenTypes_Type
19:5-10 SemanticTokenTypes_Class
19:11-14 SemanticTokenTypes_Keyword
19:15-22 SemanticTokenTypes_Variable
19:23-25 SemanticTokenTypes_Keyword
19:26-27 SemanticTokenTypes_Keyword
19:29-30 SemanticTokenTypes_Type
21:1-8 SemanticTokenTypes_Keyword
21:9-13 SemanticTokenTypes_Type
21:14-16 SemanticTokenTypes_Keyword
21:17-18 SemanticTokenTypes_TypeParameter
21:18-19 SemanticTokenTypes_Operator
21:20-21 SemanticTokenTypes_TypeParameter
22:3-6 SemanticTokenTypes_Keyword
22:7-10 SemanticTokenTypes_Variable
22:11-13 SemanticTokenTypes_Keyword
22:14-16 SemanticTokenTypes_Keyword
22:17-18 SemanticTokenTypes_Type
23:7-10 SemanticTokenTypes_Variable
23:11-13 SemanticTokenTypes_Keyword
23:14-15 SemanticTokenTypes_Keyword
23:17-18 SemanticTokenTypes_Type
25:1-6 SemanticTokenTypes_Keyword
25:7-8 SemanticTokenTypes_Type
25:9-11 SemanticTokenTypes_Keyword
25:12-13 SemanticTokenTypes_Keyword
25:14-18 SemanticTokenTypes_Keyword
26:7-8 SemanticTokenTypes_Type
26:9-11 SemanticTokenTypes_Keyword
26:12-13 SemanticTokenTypes_Keyword
26:14-18 SemanticTokenTypes_Keyword
27:1-8 SemanticTokenTypes_Keyword
27:9-14 SemanticTokenTypes_Type
28:3-6 SemanticTokenTypes_Keyword
28:7-10 SemanticTokenTypes_Variable
28:11-13 SemanticTokenTypes_Keyword
28:14-16 SemanticTokenTypes_Keyword
28:17-18 SemanticTokenTypes_Type
29:7-10 SemanticTokenTypes_Variable
29:11-13 SemanticTokenTypes_Keyword
29:14-15 SemanticTokenTypes_Keyword
29:17-18 SemanticTokenTypes_Type
31:1-6 SemanticTokenTypes_Keyword
31:7-8 SemanticTokenTypes_Type
31:9-11 SemanticTokenTypes_Keyword
31:12-13 SemanticTokenTypes_Keyword
31:14-18 SemanticTokenTypes_Keyword
32:7-8 SemanticTokenTypes_Type
32:9-11 SemanticTokenTypes_Keyword
32:12-13 SemanticTokenTypes_Keyword
32:14-18 SemanticTokenTypes_Keyword
33:1-8 SemanticTokenTypes_Keyword
33:9-14 SemanticTokenTypes_Type
33:15-16 SemanticTokenTypes_TypeParameter
33:17-18 SemanticTokenTypes_TypeParameter
34:3-6 SemanticTokenTypes_Keyword
34:7-10 SemanticTokenTypes_Variable
34:11-13 SemanticTokenTypes_Keyword
34:14-16 SemanticTokenTypes_Keyword
34:17-18 SemanticTokenTypes_Type
35:7-10 SemanticTokenTypes_Variable
35:11-13 SemanticTokenTypes_Keyword
35:14-15 SemanticTokenTypes_Keyword
35:17-18 SemanticTokenTypes_Type
37:1-6 SemanticTokenTypes_Keyword
37:7-8 SemanticTokenTypes_Type
37:9-11 SemanticTokenTypes_Keyword
37:12-13 SemanticTokenTypes_Keyword
37:14-18 SemanticTokenTypes_Keyword
37:18-19 SemanticTokenTypes_Operator
37:20-21 SemanticTokenTypes_Type
37:22-24 SemanticTokenTypes_Keyword
37:25-26 SemanticTokenTypes_Keyword
37:27-31 SemanticTokenTypes_Keyword
38:1-8 SemanticTokenTypes_Keyword
38:9-14 SemanticTokenTypes_Type
38:15-16 SemanticTokenTypes_TypeParameter
38:17-18 SemanticTokenTypes_TypeParameter
38:19-22 SemanticTokenTypes_Keyword
38:23-26 SemanticTokenTypes_Variable
38:27-29 SemanticTokenTypes_Keyword
38:30-32 SemanticTokenTypes_Keyword
38:33-34 SemanticTokenTypes_Type
38:34-35 SemanticTokenTypes_Operator
38:36-39 SemanticTokenTypes_Variable
38:40-42 SemanticTokenTypes_Keyword
38:43-44 SemanticTokenTypes_Keyword
38:46-47 SemanticTokenTypes_Type
40:1-8 SemanticTokenTypes_Keyword
40:9-13 SemanticTokenTypes_Type
40:14-16 SemanticTokenTypes_Keyword
40:17-24 SemanticTokenTypes_Type
42:1-8 SemanticTokenTypes_Keyword
42:9-16 SemanticTokenTypes_Type
42:17-18 SemanticTokenTypes_TypeParameter
42:19-21 SemanticTokenTypes_Keyword
42:22-26 SemanticTokenTypes_Type
42:27-29 SemanticTokenTypes_Keyword
42:30-31 SemanticTokenTypes_Type
44:1-6 SemanticTokenTypes_Keyword
44:7-8 SemanticTokenTypes_Type
44:9-11 SemanticTokenTypes_Keyword
44:12-13 SemanticTokenTypes_Keyword
44:14-18 SemanticTokenTypes_Keyword
45:1-8 SemanticTokenTypes_Keyword
45:9-16 SemanticTokenTypes_Type
45:17-18 SemanticTokenTypes_TypeParameter
45:19-21 SemanticTokenTypes_Keyword
45:22-26 SemanticTokenTypes_Type
45:27-29 SemanticTokenTypes_Keyword
45:30-31 SemanticTokenTypes_Type

Phase: Check
1:1-2 SemanticTokenTypes_Operator
1:3-25 SemanticTokenTypes_Interface
3:1-8 SemanticTokenTypes_Keyword
3:9-13 SemanticTokenTypes_Type
3:14-16 SemanticTokenTypes_Keyword
3:17-18 SemanticTokenTypes_TypeParameter
4:3-5 SemanticTokenTypes_Keyword
4:6-9 SemanticTokenTypes_Keyword
4:10-12 SemanticTokenTypes_Keyword
5:5-8 SemanticTokenTypes_Class
6:5-9 SemanticTokenTypes_Class
6:10-13 SemanticTokenTypes_Keyword
7:7-11 SemanticTokenTypes_Method
7:12-14 SemanticTokenTypes_Keyword
7:15-17 SemanticTokenTypes_Keyword
7:18-19 SemanticTokenTypes_TypeParameter
8:7-11 SemanticTokenTypes_Method
8:12-14 SemanticTokenTypes_Keyword
8:15-16 SemanticTokenTypes_Keyword
8:17-21 SemanticTokenTypes_Type
8:22-24 SemanticTokenTypes_Keyword
8:25-26 SemanticTokenTypes_TypeParameter
10:1-8 SemanticTokenTypes_Keyword
10:9-13 SemanticTokenTypes_Type
10:14-15 SemanticTokenTypes_TypeParameter
11:3-5 SemanticTokenTypes_Keyword
11:6-9 SemanticTokenTypes_Keyword
11:10-12 SemanticTokenTypes_Keyword
12:5-9 SemanticTokenTypes_Class
12:10-13 SemanticTokenTypes_Keyword
12:14-19 SemanticTokenTypes_Method
12:20-22 SemanticTokenTypes_Keyword
12:23-25 SemanticTokenTypes_Keyword
12:26-27 SemanticTokenTypes_TypeParameter
13:5-9 SemanticTokenTypes_Class
13:10-13 SemanticTokenTypes_Keyword
13:14-18 SemanticTokenTypes_Method
13:19-21 SemanticTokenTypes_Keyword
13:22-23 SemanticTokenTypes_Keyword
13:24-28 SemanticTokenTypes_Type
13:29-30 SemanticTokenTypes_TypeParameter
14:14-19 SemanticTokenTypes_Method
14:20-22 SemanticTokenTypes_Keyword
14:23-24 SemanticTokenTypes_Keyword
14:25-29 SemanticTokenTypes_Type
14:30-31 SemanticTokenTypes_TypeParameter
16:1-8 SemanticTokenTypes_Keyword
16:9-15 SemanticTokenTypes_Type
16:16-18 SemanticTokenTypes_Keyword
16:19-20 SemanticTokenTypes_TypeParameter
16:20-21 SemanticTokenTypes_Operator
16:22-23 SemanticTokenTypes_TypeParameter
17:3-5 SemanticTokenTypes_Keyword
17:6-9 SemanticTokenTypes_Keyword
17:10-12 SemanticTokenTypes_Keyword
18:5-9 SemanticTokenTypes_Class
18:11-14 SemanticTokenTypes_Keyword
18:15-22 SemanticTokenTypes_Method
18:23-25 SemanticTokenTypes_Keyword
18:26-28 SemanticTokenTypes_Keyword
18:29-30 SemanticTokenTypes_TypeParameter
19:5-10 SemanticTokenTypes_Class
19:11-14 SemanticTokenTypes_Keyword
19:15-22 SemanticTokenTypes_Method
19:23-25 SemanticTokenTypes_Keyword
19:26-27 SemanticTokenTypes_Keyword
19:29-30 SemanticTokenTypes_TypeParameter
21:1-8 SemanticTokenTypes_Keyword
21:9-13 SemanticTokenTypes_Type
21:14-16 SemanticTokenTypes_Keyword
21:17-18 SemanticTokenTypes_TypeParameter
21:18-19 SemanticTokenTypes_Operator
21:20-21 SemanticTokenTypes_TypeParameter
22:3-6 SemanticTokenTypes_Keyword
22:7-10 SemanticTokenTypes_Method
22:11-13 SemanticTokenTypes_Keyword
22:14-16 SemanticTokenTypes_Keyword
22:17-18 SemanticTokenTypes_TypeParameter
23:7-10 SemanticTokenTypes_Method
23:11-13 SemanticTokenTypes_Keyword
23:14-15 SemanticTokenTypes_Keyword
23:17-18 SemanticTokenTypes_TypeParameter
25:1-6 SemanticTokenTypes_Keyword
25:7-8 SemanticTokenTypes_TypeParameter
25:9-11 SemanticTokenTypes_Keyword
25:12-13 SemanticTokenTypes_Keyword
25:14-18 SemanticTokenTypes_Keyword
26:7-8 SemanticTokenTypes_TypeParameter
26:9-11 SemanticTokenTypes_Keyword
26:12-13 SemanticTokenTypes_Keyword
26:14-18 SemanticTokenTypes_Keyword
27:1-8 SemanticTokenTypes_Keyword
27:9-14 SemanticTokenTypes_Type
28:3-6 SemanticTokenTypes_Keyword
28:7-10 SemanticTokenTypes_Method
28:11-13 SemanticTokenTypes_Keyword
28:14-16 SemanticTokenTypes_Keyword
28:17-18 SemanticTokenTypes_TypeParameter
29:7-10 SemanticTokenTypes_Method
29:11-13 SemanticTokenTypes_Keyword
29:14-15 SemanticTokenTypes_Keyword
29:17-18 SemanticTokenTypes_TypeParameter
31:1-6 SemanticTokenTypes_Keyword
31:7-8 SemanticTokenTypes_TypeParameter
31:9-11 SemanticTokenTypes_Keyword
31:12-13 SemanticTokenTypes_Keyword
31:14-18 SemanticTokenTypes_Keyword
32:7-8 SemanticTokenTypes_TypeParameter
32:9-11 SemanticTokenTypes_Keyword
32:12-13 SemanticTokenTypes_Keyword
32:14-18 SemanticTokenTypes_Keyword
33:1-8 SemanticTokenTypes_Keyword
33:9-14 SemanticTokenTypes_Type
33:15-16 SemanticTokenTypes_TypeParameter
33:17-18 SemanticTokenTypes_TypeParameter
34:3-6 SemanticTokenTypes_Keyword
34:7-10 SemanticTokenTypes_Method
34:11-13 SemanticTokenTypes_Keyword
34:14-16 SemanticTokenTypes_Keyword
34:17-18 SemanticTokenTypes_TypeParameter
35:7-10 SemanticTokenTypes_Method
35:11-13 SemanticTokenTypes_Keyword
35:14-15 SemanticTokenTypes_Keyword
35:17-18 SemanticTokenTypes_TypeParameter
37:1-6 SemanticTokenTypes_Keyword
37:7-8 SemanticTokenTypes_TypeParameter
37:9-11 SemanticTokenTypes_Keyword
37:12-13 SemanticTokenTypes_Keyword
37:14-18 SemanticTokenTypes_Keyword
37:18-19 SemanticTokenTypes_Operator
37:20-21 SemanticTokenTypes_TypeParameter
37:22-24 SemanticTokenTypes_Keyword
37:25-26 SemanticTokenTypes_Keyword
37:27-31 SemanticTokenTypes_Keyword
38:1-8 SemanticTokenTypes_Keyword
38:9-14 SemanticTokenTypes_Type
38:15-16 SemanticTokenTypes_TypeParameter
38:17-18 SemanticTokenTypes_TypeParameter
38:19-22 SemanticTokenTypes_Keyword
38:23-26 SemanticTokenTypes_Method
38:27-29 SemanticTokenTypes_Keyword
38:30-32 SemanticTokenTypes_Keyword
38:33-34 SemanticTokenTypes_TypeParameter
38:34-35 SemanticTokenTypes_Operator
38:36-39 SemanticTokenTypes_Method
38:40-42 SemanticTokenTypes_Keyword
38:43-44 SemanticTokenTypes_Keyword
38:46-47 SemanticTokenTypes_TypeParameter
40:1-8 SemanticTokenTypes_Keyword
40:9-13 SemanticTokenTypes_Type
40:14-16 SemanticTokenTypes_Keyword
40:17-24 SemanticTokenTypes_Type
42:1-8 SemanticTokenTypes_Keyword
42:9-16 SemanticTokenTypes_Type
42:17-18 SemanticTokenTypes_TypeParameter
42:19-21 SemanticTokenTypes_Keyword
42:22-26 SemanticTokenTypes_Type
42:27-29 SemanticTokenTypes_Keyword
42:30-31 SemanticTokenTypes_Type
44:1-6 SemanticTokenTypes_Keyword
44:7-8 SemanticTokenTypes_TypeParameter
44:9-11 SemanticTokenTypes_Keyword
44:12-13 SemanticTokenTypes_Keyword
44:14-18 SemanticTokenTypes_Keyword
45:1-8 SemanticTokenTypes_Keyword
45:9-16 SemanticTokenTypes_Type
45:17-18 SemanticTokenTypes_TypeParameter
45:19-21 SemanticTokenTypes_Keyword
45:22-26 SemanticTokenTypes_Type
45:27-29 SemanticTokenTypes_Keyword
45:30-31 SemanticTokenTypes_Type

